nova/scheduler/filters/resize_reservation_filter.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2020 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+
+import nova.conf
+from nova.scheduler import filters
+from nova.utils import is_resize
+
+LOG = logging.getLogger(__name__)
+
+CONF = nova.conf.CONF
+
+
+class ResizeReservedRAMBaseFilter(filters.BaseHostFilter):
+
+    RUN_ON_REBUILD = False
+
+
+class ResizeReservedHostRAMFilter(ResizeReservedRAMBaseFilter):
+    """Only return hosts for resize with sufficient available RAM."""
+
+    @property
+    def reserved_percent_ram(self):
+        return CONF.filter_scheduler.reserved_ram_threshold_for_resize
+
+    def host_passes(self, host_state, request_spec):
+        instance = host_state.instances[request_spec.instance_uuid]
```

Comment:
This seems to assume, that the instance is already assigned to a host, which, as Jakob stated down below, will not work for scheduling new instances. You need another way to figure out if we're running a resize scheduling or a new build. For `rebuild`, [nova-api sets a `scheduler_hint`](https://github.com/sapcc/nova/blob/76fe6536ac413dfaf656a2899c809ad505df092f/nova/compute/api.py#L3232), which we could also set for a resize, if we wanted to ...

Given that, you need to also extend your tests to test a host that doesn't have the instance assigned to itself. It should still be possible to resize _to_ it, if it matches the conditions.